libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -201,73 +254,31 @@ public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions)
             return _collectionLink;
         }
 
-        /// <summary>
-        /// Converts the key into a DocumentID that can be used safely with CosmosDB.
-        /// The following characters are restricted and cannot be used in the Id property: '/', '\', '?', '#'
-        /// More information at https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.resource.id?view=azure-dotnet#remarks
-        /// </summary>
-        public static string SanitizeKey(string key)
-        {
-            var firstIllegalCharIndex = key.IndexOfAny(IllegalKeyCharacters);
-
-            // If there are no illegal characters return immediately and avoid any further processing/allocations
-            if (firstIllegalCharIndex == -1) return key;
-
-            // Allocate a builder that assumes that all remaining characters might be replaced to avoid any extra allocations
-            var sanitizedKeyBuilder = new StringBuilder(key.Length + (key.Length - firstIllegalCharIndex + 1) * 3);
-
-            // Add all good characters up to the first bad character to the builder first
-            for (int index = 0; index < firstIllegalCharIndex; index++)
-            {
-                sanitizedKeyBuilder.Append(key[index]);
-            }
-
-            var illegalCharacterReplacementMap = IllegalKeyCharacterReplacementMap.Value;
-
-            // Now walk the remaining characters, starting at the first known bad character, replacing any bad ones with their designated replacement value from the map
-            for (int index = firstIllegalCharIndex; index < key.Length; index++)
-            {
-                var ch = key[index];
-
-                // Check if this next character is considered illegal and, if so, append its replacement; otherwise just append the good character as is
-                if (illegalCharacterReplacementMap.TryGetValue(ch, out var replacement))
-                {
-                    sanitizedKeyBuilder.Append(replacement);
-                }
-                else
-                {
-                    sanitizedKeyBuilder.Append(ch);
-                }
-            }
-
-            return sanitizedKeyBuilder.ToString();
-        }
-
         /// <summary>
         /// Internal data structure for storing items in a CosmosDB Collection.
         /// </summary>
         private class DocumentStoreItem
         {
             /// <summary>
-            /// Sanitized Id/Key an used as PrimaryKey.
+            /// Gets or sets the sanitized Id/Key used as PrimaryKey.
             /// </summary>
```

Comment:
Is this comment still accurate? I don't see anything about "sanitized" in the code path.

Reply:
I think so. In https://github.com/southworkscom/botbuilder-dotnet/blob/a47916379f9154d1932185186b4eb8b85cda0e64/libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs#L216 is where the Id is being sanitized